tools/hash_reqs.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import os
+import sys
+import argparse
+import hashlib
+
+def expand_reqs_helper(fpath, visited):
+    if fpath in visited:
+        raise Exception("Cyclic dependency between requirements files")
+
+    curr_dir = os.path.dirname(fpath)
+    result = [] # type: List[str]
+
+    for line in open(fpath):
+        dep = line.split(" #", 1)[0].strip() # remove comments and strip whitespace
+        if dep:
+            if dep.startswith('-r'):
+                child = os.path.join(curr_dir, dep[3:])
+                result += expand_reqs_helper(child, visited.union({fpath}))
+            else:
+                result.append(dep)
+    return result
+
+def expand_reqs(fpath):
+    """
+    Returns a sorted list of unique dependencies specified by the requirements file `fpath`.
+    Removes comments from the output and recursively visits files specified inside `fpath`.
+    `fpath` can be either an absolute path or a relative path.
+    """
+    absfpath = os.path.abspath(fpath)
+    output = expand_reqs_helper(absfpath, set())
```

Comment:
Hmm, I'm not sure this set-based implementation of not visiting things multiple times will do the right thing in the case that you have a file included by multiple dependent requirements.txt files.  

It might be simpler to just implement this with a visited set + a queue of future items to visit, all run through in a loop, rather than doing this recursion (would also work to stick with recursion but make the visited set a global, and then change the check at the top to just return doing nothing rather than throw an exception when trying to visit something twice).